pilot/cmd/sidecar-injector/main.go

```diff
@@ -131,6 +134,24 @@ func patchCert() error {
 	return err
 }
 
+// patchCertLoop continually reapplies the caBundle PEM. This is required because it can be overwritten with empty
+// values if the original yaml is reapplied (https://github.com/istio/istio/issues/6069).
+func patchCertLoop(getCABundlePEM func() []byte) error {
```

Comment:
This is an improvement over the existing (broken) behavior. It will lead to some noise in api-server logs. That could be mitigated by updating PatchMutatingWebhookConfig to skip invoking PATCH if the strategic merge patch is empty.

Ideally this would be driven by watching the webhook configuration (via watch or informer) and only PATCH'ing if change is observed. Is there a follow-up tracking issue for that work item?

Reply:
There is now: https://github.com/istio/istio/issues/6451